internal/cli/service.go

```diff
@@ -364,6 +419,37 @@ func (s Service) yamlFilePathsInDirectory(dir string) ([]string, error) {
 	return paths, nil
 }
 
+// yamlFilePathsInDirectory returns any *.yml and *.yaml files in a given directory, ignoring any sub-directories.
+func (s Service) findMintDirectoryPath(configuredDirectory string) (string, error) {
+	if configuredDirectory != "" {
+		return configuredDirectory, nil
+	}
+
+	workingDirectory, err := s.FileSystem.Getwd()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to determine the working directory")
+	}
+
+	// otherwise, walk up the working directory looking at each basename
+	for {
+		workingDirHasMintDir, err := s.FileSystem.Exists(filepath.Join(workingDirectory, ".mint"))
+		if err != nil {
+			return "", errors.Wrapf(err, "unable to determine if .mint exists in %q", workingDirectory)
+		}
+
+		if workingDirHasMintDir {
+			return filepath.Join(workingDirectory, ".mint"), nil
+		}
+
+		if (workingDirectory == string(os.PathSeparator)) {
+			return "", nil
+		}
```

Comment:
Unfortunate that Go stdlib doesn't expose a "root of filesystem" function ([1](https://stackoverflow.com/a/74539327))

Reply:
Yea, I think there are other parts of this that aren't very Windows friendly anyway that we need to tackle 😐 specifically, Mint does pretty much exclusively unix path manipulation, but much of the code in here uses things like `filepath.Join` which are OS-specific (so windows would send up windows path separators even though Mint wants unix)